Client/game_sa/CFxSA.cpp

```diff
@@ -239,3 +239,93 @@ void CFxSA::TriggerFootSplash(CVector& vecPosition)
         call    dwFunc
     }
 }
+
+void CFxSA::AddParticle(eFxParticleSystems eFxParticle, CVector& vecPosition, CVector& vecDirection, float fR, float fG, float fB, float fA, bool bRandomizeColors, int iCount, float fBrightness, float fSize, bool bRandomizeSizes, float fLife)
+{
+    // Init our own FxPrtMult struct
+    FxPrtMult_c fxPrt{{fR,fG,fB,fA}, fSize, 0, fLife};
+    CVector     newDirection;
+
+    FxSystem_c* fxParticleSystem;
+
+    switch (eFxParticle)
+    {
+        case PRT_BLOOD:
+            fxParticleSystem = m_pInterface->m_fxSysBlood;
+            break;
+        case PRT_BOATSPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysBoatSplash;
+            break;
+        case PRT_BUBBLE:
+            fxParticleSystem = m_pInterface->m_fxSysBubble;
+            break;
+        case PRT_DEBRIS:
+            fxParticleSystem = m_pInterface->m_fxSysDebris;
+            break;
+        case PRT_GUNSHELL:
+            fxParticleSystem = m_pInterface->m_fxSysGunshell;
+            break;
+        case PRT_SAND:
+            fxParticleSystem = m_pInterface->m_fxSysSand;
+            break;
+        case PRT_SAND2:
+            fxParticleSystem = m_pInterface->m_fxSysSand2;
+            break;
+        case PRT_SMOKE:
+            fxParticleSystem = m_pInterface->m_fxSysSmoke;
+            break;
+        case PRT_SMOKEHUGE:
+            fxParticleSystem = m_pInterface->m_fxSysSmokeHuge;
+            break;
+        case PRT_SMOKE2:
+            fxParticleSystem = m_pInterface->m_fxSysSmoke2;
+            break;
+        case PRT_SPARK:
+            fxParticleSystem = m_pInterface->m_fxSysSpark;
+            break;
+        case PRT_SPARK2:
+            fxParticleSystem = m_pInterface->m_fxSysSpark2;
+            break;
+        case PRT_SPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysSplash;
+            break;
+        case PRT_WAKE:
+            fxParticleSystem = m_pInterface->m_fxSysWake;
+            break;
+        case PRT_WATERSPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysWaterSplash;
+            break;
+        case PRT_WHEELDIRT:
+            fxParticleSystem = m_pInterface->m_fxSysWheelDirt;
+            break;
+        case PRT_GLASS:
+            fxParticleSystem = m_pInterface->m_fxSysGlass;
+            break;
+        default:
+            fxParticleSystem = m_pInterface->m_fxSysBlood;
+    }
+
+    for (int i = 0; i < iCount; i++)
+    {
+        if (bRandomizeColors)
+        {
+            // 0x49EECB
+            fxPrt.m_color.red = (rand() % 10000) * 0.0001 * fR + 0.13;
```

Comment:
Please, use floating point numbers instead of the doubles. For example: 0.0001 -> 0.0001f.